src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp

```diff
@@ -4,13 +4,25 @@
 
 #include "functional_test_utils/core_config.hpp"
 #include "shared_test_classes/base/ov_subgraph.hpp"
+#include "utils/cpu_utils.hpp"
+#include "config.h"
 
 void CoreConfiguration(LayerTestsUtils::LayerTestsCommon* test) {
     // Within the test scope we don't need any implicit bf16 optimisations, so let's run the network as is.
     auto& configuration = test->GetConfiguration();
     if (!configuration.count(InferenceEngine::PluginConfigParams::KEY_ENFORCE_BF16)) {
         configuration.insert({InferenceEngine::PluginConfigParams::KEY_ENFORCE_BF16, InferenceEngine::PluginConfigParams::NO});
     }
+    //fp16 precision is used as default precision on ARM for non-convolution networks
+    #if defined(OV_CPU_ARM_ENABLE_FP16)
```

Comment:
I expect the impl should be fully aligned with bf16. Like if `ov::hint::inference_precision` is not explicitly set equal to fp16 in the "configuration" we force  `ov::hint::inference_precision` to be equal fp32. No necessety to have dependency on ModelType

Reply:
I believe no need to check precision value if it was set explicitly. Addressed the comment.